src/plugins.c

```diff
@@ -830,25 +1012,87 @@ static gboolean check_plugin_path(const gchar *fname)
 }
 
 
+/* Retuns NULL if this ain't a plugin,
+ * otherwise it returns the appropriate PluginProxy instance to load it */
+static PluginProxy* is_plugin(const gchar *file)
+{
+	GList *node;
+	const gchar *ext;
+
+	/* extract file extension to avoid g_str_has_suffix() in the loop */
+	ext = (const gchar *)strrchr(file, '.');
+	if (ext == NULL)
+		return FALSE;
+	/* ensure the dot is really part of the filename */
+	else if (strchr(ext, G_DIR_SEPARATOR) != NULL)
+		return FALSE;
+
+	ext += 1;
+	/* O(n*m), (m being extensions per proxy) doesn't scale very well in theory
+	 * but not a problem in practice yet */
+	foreach_list(node, active_proxies.head)
+	{
+		PluginProxy *proxy = node->data;
+		if (utils_str_casecmp(ext, proxy->extension) != 0)
```

Comment:
This test is inverted, should be `==`  It only "works" with C plugins by chance because they ship with a `.la` and for some reason loading this one also works.
Doesn't work with [geanypy port](https://github.com/kugel-/geany-plugins/commit/9c27df004f1c0906970da605d6c0ffd89d0a2fed), no python plugins show up -- and geanypy probes and tried to load all C plugins.